vulnerabilities/views.py

```diff
@@ -19,50 +19,118 @@
 from django.views import View
 from django.views.generic.edit import UpdateView
 from django.views.generic.list import ListView
+from packageurl import PackageURL
+from packaging import version
 
 from vulnerabilities import forms
 from vulnerabilities import models
+from vulnerabilities.forms import CVEForm
+from vulnerabilities.forms import PackageForm
+from vulnerablecode import settings
 
 
 class PackageSearchView(View):
     template_name = "packages.html"
+    ordering = ["version"]
 
     def get(self, request):
-        context = {"form": forms.PackageForm(request.GET or None)}
+        result_size = 0
+        context = {"debug_ui": settings.DEBUG_UI}
 
         if request.GET:
             packages = self.request_to_queryset(request)
             result_size = len(packages)
             try:
                 page_no = request.GET.get("page", 1)
+                pages = Paginator(packages, 50)
                 packages = Paginator(packages, 50).get_page(page_no)
             except PageNotAnInteger:
                 packages = Paginator(packages, 50).get_page(1)
-            packages = Paginator(packages, 50).get_page(page_no)
+            packages = pages.get_page(page_no)
+
             context["packages"] = packages
             context["searched_for"] = urlencode(
                 {param: request.GET[param] for param in request.GET if param != "page"}
             )
             context["result_size"] = result_size
 
+            if len(request.GET["type"]):
+                package_type = request.GET["type"]
+                context["package_type"] = package_type
+
+            if len(request.GET["name"]):
+                package_name = request.GET["name"]
+                context["package_name"] = package_name
+
+            if result_size == 0:
+                context = {
+                    "package_search": "The VCIO DB does not contain a record of the package you entered -- "
+                    + request.GET["name"]
+                    + ".",
```

Comment:
Use f-strings for better readability:
```
search_string = request.GET["name"]
"package_search": f"The VCIO DB does not contain a record of the package you entered: {search_string}."
```

Reply:
See new issue https://github.com/nexB/vulnerablecode/issues/877